data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/export/MongoDBExportPartitionSupplier.java

```diff
@@ -111,11 +115,15 @@ private PartitionIdentifierBatch buildPartitions(final ExportPartition exportPar
                             .builder()
                             .withPartitionKey(String.format(MONGODB_PARTITION_KEY_FORMAT, collectionDbName, gteValueString, lteValueString, gteClassName, lteClassName))
                             .build());
+                    documentDBAggregateMetrics.getExportPartitionQueryCount().increment();
 
                     if (isLastBatch) {
                         break;
                     }
 
+                    // extend the ownership of the partition
+                    enhancedSourceCoordinator.saveProgressStateForPartition(exportPartition, null);
```

Comment:
Will this spam the source coordinator or should there be some time difference or condition for us to increase ownership timeout?

Reply:
It will only make calls to source coordinator when query partitions are created. For very large collection it take long time to scan entire collection, so we need to keep ownership of the partition. The calls to source coordinator must be max in 10's per second. I ran the test locally and I see version bump to 150 in 4-5 mins.